support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/validator/token/OAuth20ClientCredentialsGrantTypeTokenRequestValidator.java

```diff
@@ -5,23 +5,32 @@
 import org.apereo.cas.support.oauth.util.OAuth20Utils;
 import org.apereo.cas.support.oauth.web.endpoints.OAuth20ConfigurationContext;
 
+import lombok.extern.slf4j.Slf4j;
 import lombok.val;
 import org.apache.commons.lang3.StringUtils;
 import org.pac4j.core.context.WebContext;
+import org.pac4j.jee.context.JEEContext;
 
 /**
  * This is {@link OAuth20ClientCredentialsGrantTypeTokenRequestValidator}.
  *
  * @author Misagh Moayyed
  * @since 5.3.0
  */
+@Slf4j
 public class OAuth20ClientCredentialsGrantTypeTokenRequestValidator extends OAuth20PasswordGrantTypeTokenRequestValidator {
     public OAuth20ClientCredentialsGrantTypeTokenRequestValidator(final OAuth20ConfigurationContext configurationContext) {
         super(configurationContext);
     }
 
     @Override
     public boolean supports(final WebContext context) {
+        val queryString = ((JEEContext) context).getNativeRequest().getQueryString();
```

Comment:
You should use `OAuth20RequestParameterResolver#resolveRequestParameter()` and you will get back an optional, like the few lines below do with resolving grant_type. Furthermore, you should only do this if the grant-type is in fact client_credentials after the check below. That is, the support check should be:

1. This is a client-credentials request
2. AND, the request does not contain a client secret

Only then the request validator can support this request.

Please also add unit tests.

PS There are many more puppeteer tests that likely do the same thing. Please verify those. (You will likely see test failures in many other areas).

Reply:
Sure. Will do

Reply:
Trying to account for your feedback, I'm not sure to get it right: first, I should check the `client_credentials`grant type and then the `client_secret` parameter. That's clear.

Reply:
Though, I'm not sure to understand why I should use `OAuth20RequestParameterResolver#resolveRequestParameter()`...

Reply:
The `resolveRequestParameter` method calls: `context.getRequestParameters().get(name)` where `context` is a `JEEContext` which internally calls: `HttpServletRequest.getParameterMap()`  which returns parameters in the URL but also in the body, doesn't it?

Reply:
You were right: many other tests use the client_credentials grant type: fixing...

Reply:
Would you be OK with the following logic: `OAuth20ClientCredentialsGrantTypeTokenRequestValidator.supports` would call `OAuth20RequestParameterResolver.ensureClientSecretIsNotInTheQueryString`?

Reply:
@apereocas-bot runci

Reply:
I just updated the PR. Your comments should have properly been taken into account: order of checks, unit tests, integration tests, new method `isParameterOnQueryString`.

Reply:
Perfect, thank you.